javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/JavaParserFacade.java

```diff
@@ -614,25 +591,47 @@ protected Node findContainingTypeDeclOrObjectCreationExpr(Node node) {
         }
         return findContainingTypeDeclOrObjectCreationExpr(parent);
     }
-    
-    /**
-     * Where a node has an interface/class/enum declaration -- or an object creation expression in an inner class
-     * references an outer class -- as its ancestor, return the declaration corresponding to the class name specified.
-     */
-    protected Node findContainingTypeDeclOrObjectCreationExpr(Node node, String className) {
-        if (node instanceof ClassOrInterfaceDeclaration && ((ClassOrInterfaceDeclaration) node).getFullyQualifiedName().get().equals(className)) {
-            return node;
-        }
-        if (node instanceof EnumDeclaration) {
-            return node;
-        }
-        Node parent = demandParentNode(node);
-        if (parent instanceof ObjectCreationExpr && !((ObjectCreationExpr) parent).getArguments().contains(node)) {
-            return parent;
+
+    protected TypeDeclaration<?> findContainingTypeDecl(Node node, String className) {
```

Comment:
The javadoc for both `findContainingTypeDecl` and `findContainingTypeDeclOrObjectCreationExpr` has been removed - is that deliberate?

Reply:
Ah I see, maybe we can keep the removed method with its java docs, and add java docs to both new methods.

Reply:
Ah I see, maybe we can keep the removed method with its java docs, and add java docs to both new methods.